crates/modelardb_server/src/storage/uncompressed_data_buffer.rs

```diff
@@ -198,14 +199,31 @@ impl fmt::Debug for UncompressedInMemoryDataBuffer {
 
 #[async_trait]
 impl UncompressedDataBuffer for UncompressedInMemoryDataBuffer {
-    /// Finish the array builders and return the data in a structured [`RecordBatch`].
+    /// Finish the array builders and return the data in a [`RecordBatch`] sorted by time.
     async fn record_batch(&mut self) -> Result<RecordBatch, ParquetError> {
-        let timestamps = self.timestamps.finish();
+        // SortColumn requires that the values are in an Arc but this is not needed for take.
+        let timestamps = Arc::new(self.timestamps.finish());
         let values = self.values.finish();
 
+        // sort_to_indices is used instead of lexsort as it is unclear how lexsort sorts multiple
```

Comment:
Maybe `lexsort_to_indices()` instead to fix the function name and make it clear that it is a function.

Reply:
If this is changed, also change the other instance of this and change all `lexsort` to `lexsort()` and all `take` to `take()`.

Reply:
Fixed.